core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -294,43 +298,43 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
 
   def updateErrorMetrics(apiKey: ApiKeys, errors: collection.Map[Errors, Integer]) {
     errors.foreach { case (error, count) =>
-      RequestMetrics.markErrorMeter(apiKey.name, error, count)
+      requestChannelMetrics.markErrorMeter(apiKey.name, error, count)
     }
   }
 
   def shutdown() {
     requestQueue.clear()
+    requestChannelMetrics.shutdown()
   }
 
   def sendShutdownRequest(): Unit = requestQueue.put(ShutdownRequest)
 
 }
 
-object RequestMetrics {
+class RequestChannelMetrics {
```

Comment:
Maybe this can be an inner class in the `RequestChannel` object called `Metrics` or `MetricsMap`.